File: src/main/java/pr/eleks/we_at_her/services/view/ViewService.java
package pr.eleks.we_at_her.services.view;

import org.springframework.web.util.UriComponentsBuilder;
import pr.eleks.we_at_her.dto.BlogPostDto;
import pr.eleks.we_at_her.dto.CityDto;
import pr.eleks.we_at_her.dto.UserDto;
import pr.eleks.we_at_her.dto.weather.WeatherSampleDto;
import pr.eleks.we_at_her.exceptions.PropertyNotFoundException;

import java.util.List;

public interface ViewService {
    UriComponentsBuilder hostUriBuilder() throws PropertyNotFoundException;

    List<WeatherSampleDto> getAllWeatherSamples() throws PropertyNotFoundException;

    List<CityDto> getAllCities() throws PropertyNotFoundException;

    UserDto createUser(UserDto userDto) throws PropertyNotFoundException;

    List<BlogPostDto> getAllBlogPosts() throws PropertyNotFoundException;

    UserDto activateUser(String userUuid) throws PropertyNotFoundException;
}

File: src/main/kotlin/pr/eleks/we_at_her/services/view/ViewService.kt
package pr.eleks.we_at_her.services.view

import org.springframework.web.util.UriComponentsBuilder
import pr.eleks.we_at_her.dto.BlogPostDto
import pr.eleks.we_at_her.dto.CityDto
import pr.eleks.we_at_her.dto.UserDto
import pr.eleks.we_at_her.dto.weather.WeatherSampleDto
import pr.eleks.we_at_her.exceptions.PropertyNotFoundException

interface ViewService {
    fun hostUriBuilder(): UriComponentsBuilder
    fun getAllWeatherSamples(): List<WeatherSampleDto?>
    fun getAllCities(): List<CityDto?>
    fun createUser(userDto: UserDto): UserDto?
    fun getAllBlogPosts(): List<BlogPostDto?>
    fun activateUser(userUuid: String): UserDto?
    fun createNewBlogPost(blogPostDto: BlogPostDto): BlogPostDto?
}